### PR TITLE
Better fp16 detection to speed up training

### PR DESCRIPTION
**Pull request by kalomaze**
Fp16 is being detected in a terrible way right now that is causing false positives and not properly updating

### CODE REVIEW

**Review by alexlnkp — APPROVED** [review]
gogogo

File: config.py [hunk 1]
```diff
@@ -84,9 +137,9 @@ def device_config(self) -> tuple:
             ):
                 print("Found GPU", self.gpu_name, ", force to fp32")
                 self.is_half = False
-                use_fp32_config()
             else:
                 print("Found GPU", self.gpu_name)
+                use_fp32_config()
```
Comment (fumiama) [inline config.py:142]: The function name is still `use_fp32_config` so you might want to change its name to `use_fp16_config` for example.

**Review by fumiama — CHANGES_REQUESTED** [review]
I think you can change the behavior of `device_config` instead of `use_fp32_config`. You can see that that function just determining whether to use fp16 by gpu's name.

**Comment by RVC-Boss** [comment]
I have fixed it. I will temporarily close this PR for now.
https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/issues/811

**Pull request closed without merging**